control/controlhttp/http_test.go

```diff
@@ -176,17 +176,17 @@ func testControlHTTP(t *testing.T, param httpTestParam) {
 		defer cancel()
 	}
 
-	a := dialParams{
-		host:              "localhost",
-		httpPort:          strconv.Itoa(httpLn.Addr().(*net.TCPAddr).Port),
-		httpsPort:         strconv.Itoa(httpsLn.Addr().(*net.TCPAddr).Port),
-		machineKey:        client,
-		controlKey:        server.Public(),
-		version:           testProtocolVersion,
+	a := Dialer{
```

Comment:
&Dialer

(to stay in same type as receiver methods and prevent accidental value copies)

Reply:
Addressed in #5661